keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java

```diff
@@ -0,0 +1,572 @@
+     * System.out.printf("Key is created with name %s and id %s \n", retKey.name(), retKey.id());
+     * </pre>
+     *
+     * @param name The name of the key being created.
+     * @param keyType The type of key to create. For valid values, see {@link JsonWebKeyType JsonWebKeyType}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    public Response<Key> createKey(String name, JsonWebKeyType keyType) {
+        KeyRequestParameters parameters = new KeyRequestParameters().kty(keyType);
+        return service.createKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Creates a new RSA key and  stores it in the key vault. The create rsa key operation can be used to create any rsa key type in
+     * key vault. If the named key already exists, Azure Key Vault creates a new version of the key. It requires the {@code keys/create} permission.
+     *
+     * <p>The {@code rsaKeyCreateConfig} is required and its fields {@link RSAKeyCreateConfig#name() name} and {@link RSAKeyCreateConfig#keyType() key type} cannot
+     * be null. The {@link RSAKeyCreateConfig#keySize() key size}, {@link RSAKeyCreateConfig#expires() expires} and {@link RSAKeyCreateConfig#notBefore() notBefore} values
+     * in {@code rsaKeyCreateConfig} are optional. If not specified, no values are set for the fields. The {@link RSAKeyCreateConfig#enabled() enabled} field is
+     * set to true by Azure Key Vault, if not specified.</p>
+     *
+     * <p>The {@link RSAKeyCreateConfig#keyType() keyType} indicates the type of key to create. Possible values include: {@link JsonWebKeyType#RSA RSA},
+     * {@link JsonWebKeyType#RSA_HSM RSA-HSM}.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Creates a new RSA key with size 2048 which activates in one day and expires in one year. Prints out the details of the created key.</p>
+     * <pre>
+     * RSAKeyCreateConfig rsaKeyCreateConfig = new RSAKeyCreateConfig("keyName", JsonWebKeyType.RSA)
+     *    .keySize(2048)
+     *    .notBefore(OffsetDateTime.now().plusDays(1))
+     *    .expires(OffsetDateTime.now().plusYears(1));
+     *
+     * Key rsaKey = keyClient.createRSAKey(rsaKeyCreateConfig).value();
+     * System.out.printf("Key is created with name %s and id %s \n", rsaKey.name(), rsaKey.id());
+     * </pre>
+     *
+     * @param rsaKeyCreateConfig The key configuration object containing information about the rsa key being created.
+     * @throws NullPointerException if {@code rsaKeyCreateConfig} is {@code null}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the {@link Key created key}.
+     */
+    public Response<Key> createRSAKey(RSAKeyCreateConfig rsaKeyCreateConfig) {
```

Comment:
I think the capitalization of EC and RSA should still follow camel case when used in method and type names.  createRsaKey and RsaKeyCreateConfig.  This applies to all methods and types with RSA or EC

Reply:
updated.